lib/hacluster.pm

```diff
@@ -228,8 +229,36 @@ sub ha_export_logs {
 
 sub check_cluster_state {
     assert_script_run "$crm_mon_cmd";
+    assert_script_run "$crm_mon_cmd | grep -i 'no inactive resources'";
     assert_script_run 'crm_mon -1 | grep \'partition with quorum\'';
     assert_script_run 'crm_mon -s | grep "$(crm node list | wc -l) nodes online"';
+    assert_script_run 'crm_verify -LV';
+}
+
+# Wait for resources to be started
+sub wait_until_resources_started {
+    my @cmds      = ('crm cluster wait_for_startup', "$crm_mon_cmd | grep -i 'no inactive resources'");
+    my $timeout   = 120;
+    my $starttime = time;
+    my $ret       = undef;
+
+    # Execute each comnmand to validate that the cluster is running
+    # This can takes time, so a loop is a good idea here
+    foreach my $cmd (@cmds) {
+        # Check for cluster/resources status and exit loop when needed
+        while ($ret = script_run "$cmd") {
+            # Otherwise wait a while if timeout is not reached
+            my $timerun = time - $starttime;
+            if ($timerun < $timeout) {
```

Comment:
As of now I see that timeout is for all commands in the array, could you please put a comment somewhere. I like this approach. As below message in die says can be interpreted that single command failed in 120 seconds. So maybe you wanted to set `$starttime = time` before each iteration.

Reply:
Fixed!

Reply:
Thanks!